pkg/controller/master/nodedrain/nodedrain_controller.go

```diff
@@ -441,25 +441,35 @@ func (ndc *ControllerHandler) CheckVMISchedulingRequirements(originalNode *corev
 	if err != nil {
 		return nil, fmt.Errorf("error listing nodes from nodeCache: %v", err)
 	}
+	var validNodes []*corev1.Node
```

Comment:
Yesterday when looking the previous PR, I was wondering why the `originalNode` is not skipped in certain filter.

Now, let's check `isNodeReady` in this block, filter any none-ready nodes, and the following `filterNodesForNodeSelector` implicitly needs this condition, but it was not checked before.